src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -101,18 +121,27 @@ public override IEnumerable<StringGetter[]> GetSignatures()
         private partial class UserDefinitionsNameResolver : INameResolver
         {
             private readonly INameResolver _globalNameResolver;
-            private readonly INameResolver _parameterNamedResolver;
+            private readonly Dictionary<string, UDFArg> _args;
             private readonly INameResolver _functionNameResolver;
 
-            public static INameResolver Merge(INameResolver globalNameResolver, RecordType parametersRecordType, INameResolver functionNameResolver)
+            public static INameResolver Create(INameResolver globalNameResolver, ISet<UDFArg> args, INameResolver functionNameResolver)
             {
-                return new UserDefinitionsNameResolver(globalNameResolver, ReadOnlySymbolTable.NewFromRecord(parametersRecordType), functionNameResolver);
+                return new UserDefinitionsNameResolver(globalNameResolver, args, functionNameResolver);
             }
 
-            private UserDefinitionsNameResolver(INameResolver globalNameResolver, INameResolver parameterNamedResolver, INameResolver functionNameResolver = null)
+            private UserDefinitionsNameResolver(INameResolver globalNameResolver, ISet<UDFArg> args, INameResolver functionNameResolver = null)
             {
                 this._globalNameResolver = globalNameResolver;
-                this._parameterNamedResolver = parameterNamedResolver;
+                this._args = new Dictionary<string, UDFArg>();
+
+                foreach (UDFArg arg in args)
+                {
+                    if (!this._args.ContainsKey(arg.VarIdent.Name.Value))
```

Comment:
Should this mark it as erroneous if there are colliding keys?

Reply:
or where should that be handled.

Reply:
Then why check ContainsKey at all? 
Why not just
`this._args = args.ToDictionary(arg => arg.VarIdent.Name.Value, arg => arg)`

Reply:
By handling I meant, we see that there is a duplicate and add an error on the node but we don't remove that from the args array.